rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -436,6 +396,34 @@ fn write(fd: RawFd, buf: &[u8]) -> io::Result<usize> {
     }
 }
 
+async fn add_route(route: &IpNetwork, idx: u32, handle: &Handle) {
+    let res = match route {
+        IpNetwork::V4(ipnet) => make_route_v4(idx, handle, *ipnet).execute().await,
+        IpNetwork::V6(ipnet) => make_route_v6(idx, handle, *ipnet).execute().await,
+    };
+
+    match res {
+        Ok(_) => {}
+        Err(NetlinkError(err)) if err.raw_code() == FILE_ALREADY_EXISTS => {}
+        // TODO: we should be able to surface this error and handle it depending on
+        // if any of the added routes succeeded.
+        Err(err) => {
+            tracing::error!(%route, "failed to add route: {err:#?}");
```

Comment:
I'd suggest to not use debug-pretty print as it is quite verbose and spans multiple lines.

```suggestion
            tracing::error!(%route, "failed to add route: {err}");
```